flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -420,7 +420,7 @@ public void testCancellationOfCanceledTerminalDoesNotThrowException() throws Exc
                                         .build())));
 
         // wait for job to finish
-        dispatcherGateway.requestJobResult(jobId, TIMEOUT).get();
+        dispatcher.getJobTerminationFuture(jobId, TIMEOUT).get();
```

Comment:
Switching from `requestJobResult` to `getJobTerminationFuture` we're now always waiting for the cleanup to be done before triggering whatever logic we want to test afterwards. That means, that the `JobManagerRunner` is always deregistered. I'm wondering whether we should add a test to `DispatcherTest` that covers the codepath of `Dispatcher#requestJobStatus` where the `JobManagerRunner` is not unregistered, yet. WDYT? 🤔

Reply:
That's kinda covered by `testRequestMultipleJobDetails_returnsRunningOverSuspendedJob`; there one of the mock JMRunners doesn't terminate and implements the required methods.

Reply:
I'm not sure about that. `Dispatcher#requestJobResult` is not necessarily triggered before the job submission goes through in [DispatcherTest:969](https://github.com/apache/flink/blob/1cb80e6d1ed0331ba662e481705b6b716f29b26a/flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java#L969). Therefore, we might endup in the code branch of `requestJobResult` again where the `Dispatcher` accesses the `ExecutionGraphInfoStore`.

I was thinking about something like this: https://gist.github.com/XComp/6f082c15ce10ceae3f743f84b6bc6903

Reply:
true, I was thinking of `requestMultipleJobDetails()`. In any case the PR doesn't reduce test coverage; I only touched tests that used a JMRunner which would've failed if the job result were to be requested.